conan/api/subapi/profiles.py

```diff
@@ -29,12 +29,26 @@ def get_default_build(self):
         return loader.get_default_build()
 
     def get_profiles_from_args(self, args):
-        build = [self.get_default_build()] if not args.profile_build else args.profile_build
-        host = [self.get_default_host()] if not args.profile_host else args.profile_host
-        profile_build = self.get_profile(profiles=build, settings=args.settings_build,
-                                         options=args.options_build, conf=args.conf_build)
-        profile_host = self.get_profile(profiles=host, settings=args.settings_host,
-                                        options=args.options_host, conf=args.conf_host)
+        all_profiles = [] if not args.profile_all else args.profile_all
```

Comment:
```suggestion
        all_profiles = args.profile_all or []
```
If the comparison is ``not profile_all``, short circuit ``or`` might be easier to read

Reply:
nice python-fu, you're right :)